libs/map-editor/src/Commands/Area/CreateAreaCommand.ts

```diff
@@ -0,0 +1,34 @@
+import { GameMap } from '../../GameMap/GameMap';
+import { AreaType, ITiledMapRectangleObject } from '../../types';
+import { Command } from "../Command";
+import { DeleteAreaCommandConfig } from './DeleteAreaCommand';
+
+export interface CreateAreaCommandConfig {
+    type: "CreateAreaCommand";
+    areaObjectConfig: ITiledMapRectangleObject;
+}
+
+export class CreateAreaCommand extends Command {
+    private areaConfig: ITiledMapRectangleObject;
+
+    private gameMap: GameMap;
+    
+    constructor(gameMap: GameMap, config: CreateAreaCommandConfig) {
+        super();
+        this.gameMap = gameMap;
+        this.areaConfig = {
+            ...config.areaObjectConfig,
+        };
+    }
+
+    public execute(): CreateAreaCommandConfig {
+        this.gameMap.getGameMapAreas().addArea(this.areaConfig, AreaType.Static);
+        this.gameMap.incrementNextObjectId();
```

Comment:
Instead of calling `incrementNextObjectId` here, you could call it directly from the `GameMapAreas.addArea` method.
This is safer. If in the future we call `GameMapAreas.addArea` from another place, the object id will be automatically dealt with.